File: 1-Patron/python/Ultimate.py
class Ultimate(object):

    def __init__(self, nombre, danio, nivel):
        self.__nombre = nombre
        self.__danio = danio
        self.__nivel = nivel

    @property
    def nombre(self):
        return self.__nombre

    @nombre.setter
    def nombre(self, new_nombre):
        self.__nombre = new_nombre

    @property
    def danio(self):
        return self.__danio

    @danio.setter
    def danio(self, new_danio):
        self.__danio = new_danio

    @property
    def nivel(self):
        return self.__nivel

    @nivel.setter
    def nivel(self, new_nivel):
        self.__nivel = new_nivel

    def incNivel(self, incremento):
        self.__nivel += incremento
        return self.__nivel

File: 2-Patron/Java/Ultimate.java
package abstractmasterwhole;
public class Ultimate {
    private String nombre;
    private int dano;
    private int nivel;
    
    public Ultimate(String nombre, int dano, int nivel){
        this.nombre = nombre;
        this.nivel = nivel;
        this.dano = dano;
    }
    
    public String getNombre(){
        return this.nombre;
    }
    public void setNombre(String nombre){
        this.nombre = nombre;
    }
    public int getNivel(){
        return this.nivel;
    }
    public void setNivel(int nivel){
        this.nivel = nivel;
    }
    public int getDano(){
        return this.dano;
    }
    public void setDano(int dano){
        this.dano = dano;
    }
    public int incNivel(int incremento){
        this.nivel += incremento; 
        return (this.nivel);
    }
    
}
